examples/xdp/xdp.c

```diff
@@ -0,0 +1,74 @@
+// +build ignore
+
+#include "bpf_endian.h"
+#include "common.h"
+
+char __license[] SEC("license") = "Dual MIT/GPL";
+
+const int MAX_MAP_ENTRIES = 16;
+
+/* Define an LRU hash map for storing packet count by source IPv4 address */
+struct bpf_map_def SEC("maps") xdp_stats_map = {
+	.type        = BPF_MAP_TYPE_LRU_HASH,
+	.key_size    = sizeof(__u32), // source IPv4 address
+	.value_size  = sizeof(__u32), // packet count
+	.max_entries = MAX_MAP_ENTRIES,
+};
+
+/*
+Attempt to parse the IPv4 source address from the packet.
+Returns 0 if there is no IPv4 header field; otherwise returns non-zero.
+*/
+static __always_inline int parse_ip_src_addr(struct xdp_md *ctx, __u32 *ip_src_addr) {
+	void *data_end = (void *)(long)ctx->data_end;
+	void *data     = (void *)(long)ctx->data;
+	void *pos      = data;
+
+	// First, parse the ethernet header.
+	struct ethhdr *eth = pos;
+	int hdrsize        = sizeof(*eth);
+	if (pos + hdrsize > data_end) {
+		return 0;
+	}
+
+	pos += hdrsize;
+	if (eth->h_proto != bpf_htons(ETH_P_IP)) {
+		// The protocol is not IPv4, so we can't parse an IPv4 source address.
+		return 0;
+	}
+
+	// Then parse the IP header.
+	struct iphdr *ip = pos;
+	hdrsize          = sizeof(*ip);
+	if (pos + hdrsize > data_end) {
+		return 0;
+	}
+
+	// Return the source IP address in host byte order.
+	*ip_src_addr = (__u32)bpf_ntohl(ip->saddr);
+	return 1;
+}
+
+SEC("xdp_prog")
```

Comment:
```suggestion
SEC("xdp")
```
Not sure how big of an issue it is, but libbpf [has decided](https://github.com/libbpf/libbpf/wiki/Libbpf:-the-road-to-v1.0#stricter-and-more-uniform-bpf-program-section-name-sec-handling) this should just be `xdp`.

Reply:
updated!